### ISSUE TITLE
example ogbg-mol-esol_batch with GINconv please

### ISSUE THREAD

**Issue by thegodone** [danielegrattarola/spektral#115]
I am trying to implement ESOL model but with GINconv and got lot of issues:

So can you give a clear example on how to merge tud_disjoint & ogbg-mol-esol_batch examples please ?
 - is A must be sparse ?
 - ...

**Comment by danielegrattarola** [comment]
Hi, 

have you looked at [this example](https://github.com/danielegrattarola/spektral/blob/master/examples/graph_prediction/ogbg-mol-hiv_disjoint.py)?
With the following changes

- use 'ogbg-molesol'
- change `EdgeConditionedConv` to `GINConv`
- copy-paste the evaluator from the `ogbg-mol-esol_batch` example

you should be good. 

Keep in mind that ogbg-molesol has edge attributes but GINConv cannot process them. So you should either
1. Keep everything as-is and simply not use the `E_in` input layer (this is the fastest way in terms of coding). 
2. Change the code to not use edge attributes (basically remove any object/variable that starts with `E_`).

Also, yes. The adjacency matrix for GINConv should be sparse. 

Cheers

**Comment by thegodone** [comment]
Thanks Daniele,

Model compiles and converges now.

But after the training part, my evaluation is strange cause y_pred are changing each time I run it and the value (rmse) is not very good compare to the model validation mse:

I have to use this method cause I need the I_ values in the models: 
################################################################################
# EVALUATE MODEL
################################################################################
print('Testing model')
evaluator = Evaluator(name=dataset_name)
y_pred = []
batches_test = batch_iterator([X_te, A_te], batch_size=batch_size)
for b in batches_test:
    X_, A_, I_ = numpy_to_disjoint(*b)
    A_ = ops.sp_matrix_to_sp_tensor(A_)
    p = model([X_, A_, I_], training=False)
    y_pred.append(p.numpy())

y_pred = np.vstack(y_pred)
print(y_pred)
model_loss = loss_fn(y_te, y_pred)
ogb_score = evaluator.eval({'y_true': y_te, 'y_pred': y_pred})

print('Done. Test loss: {:.4f}. rmse: {:.2f}'.format(model_loss, ogb_score['rmse']))

 Any idea why the training=False parameters looks not used, or I made a mistake ?

**Comment by thegodone** [comment]
another related question on GIN code
    def call(self, inputs):
        X, A, E = self.get_inputs(inputs)
        output = self.mlp((1.0 + self.eps) * X + self.propagate(X, A, E))

why to pass E there ?

**Comment by danielegrattarola** [comment]
Good to know that it works, although I am not sure what is giving you different predictions every time. 
`training=False` should be picked up automatically by TF to disable dropout for inference. What happens if you remove Dropout from the model?

As for why there is `E` in GinConv: I honestly don't remember why I thought it would be a good idea to keep it there, since it will be `None` anyway. 
You can safely ignore it without problems, I'll fix it in a later release.

Cheers

**Comment by thegodone** [comment]
same issue without Dropout. I was suspect it but it does not solve the
problem

**Comment by danielegrattarola** [comment]
sorry, I forgot about this issue. 
How different are the embeddings? Can you run a `np.allclose(embedding_1, embedding_2)`? 
It could be that the non-determinism of some TF ops is leading to different results, but they shouldn't be that far if the input is the same. 

Cheers

**Comment by thegodone** [comment]
Could you make a real example on mol-esol with the new tf 2.0 and last version ?
